### DISCUSSION TITLE
NodeMCU ESP8266 Development Board with 0.96 Inch OLED - Screen not working?!

### DISCUSSION THREAD

**Discussion by Cybertr0n313** [spacehuhn/esp8266_deauther#1661] [Q&A]
Cheers, I've tried several bins and I can't get any of them to work with the OLED screen. Or, is there something I need to change in the settings?  The brand is: ideashark 
1. Which is the proper bin to use? 
2. Is there something I need to change in the settings?
Please advise

**Comment by Eney888** [comment]
Good day. The same thing. The screen on the NodeMCU board with a soldered display does not work. Tried many firmwares and none of them work! Tried Arduino IDE programming. In section A_config.h you need to change the display settings and the wrong pins are specified there!
there are 5 and 4, and it should be 12 and 14. Although this did not help either, because Wifi does not work when programming ArduinoIDE

**Reply by modhack2003** [reply]
Why is the wifi not working when we are flashing it using Arduino ? My display is also not working, but  I check it with adafruit ssd1306 library examples working fine.

**Comment by kendalpercimoney** [comment]
I was able to get it to work by using the following code setup: 
[
#include "SSD1306Wire.h"
#include "ESP8266WiFi.h"

// Initialize the OLED display using Arduino Wire:
SSD1306Wire display(0x3c, D6, D5); // reversed!

void setup() {
  
  // Initialising the UI will init the display too.
  display.init();

  //display.flipScreenVertically();
  display.setFont(ArialMT_Plain_10);
  display.setTextAlignment(TEXT_ALIGN_LEFT);
  display.drawString(0, 0, "beginning..." );
  display.display();
}


void loop() {
  // clear the display
  display.clear();
  int height = 10;
  
  display.drawString(0, 12, "Screen is working!");
  display.display();

  // Wait a bit before scanning again
  delay(5000);
}](https://github.com/kendalpercimoney/esp8266-playground/blob/main/Screen_Simplified/Screen_Simplified.ino)

I installed the library http://arduino.esp8266.com/stable/package_esp8266com_index.json

**Comment by devrimq** [comment]
Bunu denedim oldu bendeki problem yanlış pin sorunu   >>>>>  https://github.com/peff74/esp8266_OLED_HW-364A  

HW-364A / HW-364B, 0,96 inç OLED ekrana sahip, oldukça kullanışlı ve uygun fiyatlı bir ESP8266 işlemcisidir.
Tüm bilinen platformlarda satılmaktadır.
Bazı satıcılar yanlış teknik veriler sağlıyor, bu da bazen cihazın çalışır hale getirilmesini zorlaştırıyor...
SDA ve SCL'nin yer değiştirebileceğini unutmayın!!!
Teknik özellikler
SDA, D6 / GPIO14'tür.
SCL, D5 / GPIO12'dir.
Adres 0x3C'dir.   


#define OLED_RESET     -1   // Reset pin # (or -1 if sharing Arduino reset pin)
#define SCREEN_ADDRESS 0x3C // If not work please scan the bus
#define OLED_SDA 14         // D6
#define OLED_SCL 12         // D5
